### ISSUE TITLE
x/tools/gopls/internal/regtest/diagnostics: failures in `TestEditGoDirectiveWorkspace`

### ISSUE THREAD

**Issue by bcmills** [golang/go#52040]
`TestEditGoDirectiveWorkspace` was added 2022-03-17 in [CL 393534](https://go.dev/cl/393534) (attn @suzmue; CC @findleyr). Unfortunately, it has failed several times since then:

```
--- FAIL: TestEditGoDirectiveWorkspace (534.58s)
    --- FAIL: TestEditGoDirectiveWorkspace/experimental (534.41s)
        diagnostics_test.go:2247: waiting on:
            	Unmet: empty diagnostics for "main.go"
            
            err:context deadline exceeded
            
            state:
            #### log messages (see RPC logs for full text):
            	Info: "2022/03/18 21:47:13 go env for /tmp/workdir-host-l...
            	Info: "2022/03/18 21:47:14 go/packages.Load\n\tsnapshot=0...
            	Error: "2022/03/18 21:47:14 building workspace mod file: ...
            	Error: "2022/03/18 21:47:14 errors loading workspace: no ...
            	Info: "2022/03/18 21:47:44 background imports cache refre...
            	Info: "2022/03/18 21:47:44 background refresh finished af...
            
            #### diagnostics:
            	main.go (version 0):
            		(2, 7): type parameters require go1.18 or later
            		(2, 9): undeclared name: any (requires version go1.18 or later)
            	go.mod (version 0):
            		(0, 0): no module path for file:///tmp/workdir-host-linux-amd64-wsl/tmp/gopls-regtest-4097331373/TestEditGoDirectiveWorkspace/experimental/work/go.mod
            
            #### outstanding work:
            	Error loading workspace: 0.00
            #### completed work:
            	Load: 1
            	Setting up workspace: 1
            	diagnosing initial workspace load: 1
            	diagnosing files changed on disk: 1
        runner.go:324: closing editor: Shutdown: context deadline exceeded
FAIL
FAIL	golang.org/x/tools/gopls/internal/regtest/diagnostics	577.903s
```

Given the variety of platforms on which it has failed, I suspect that either the test or the implementation under test is racy.

`greplogs --dashboard -md -l -e 'FAIL: TestEditGoDirectiveWorkspace' --since=2022-01-01`

[2022-03-25T20:23:27-e309672-3dac914/linux-amd64-wsl](https://build.golang.org/log/823b576546268413f4fdb44710da022ff3a574f4)
[2022-03-22T21:30:29-87a8611-ac3efc8/linux-amd64-wsl](https://build.golang.org/log/c93c99868fc9e763b46ef4b20a14bc5e1f6cad30)
[2022-03-18T23:20:51-5ea13d0-7ca6902/aix-ppc64](https://build.golang.org/log/3f547a494fdc1d73db97fa83aedf58e39dd6462a)
[2022-03-18T19:32:40-5ea13d0-a682a5c/plan9-amd64-0intro](https://build.golang.org/log/c6e2f88c2f9e9f03f2c8b650ed1d42592af72726)
[2022-03-18T13:39:41-5ea13d0-4aa1efe/linux-amd64-wsl](https://build.golang.org/log/d7bffe5cfda856c45cb3fb0e6b7ccff5ac2dbdae)

**Comment by findleyr** [comment]
Thanks, I believe this is a gopls bug: I have noticed that this behavior is flaky in practice.

**Comment by findleyr** [comment]
This is a duplicate of #51930

**Comment by bcmills** [comment]
Ah, I think that issue didn't show up in my issue search because it refers to `TestEditGoDirective` rather than `TestEditGoDirectiveWorkspace`. (But if we believe the underlying bug is the same for both, I'm happy with consolidating the issues.)
